### ISSUE TITLE
improvement - relationships - touch source or destination when object is updated

### ISSUE THREAD

**Issue by jeremiegirault** [yapstudios/YapDatabase#123]
I refer to the documentation of touchObjectForKey in View extension :
````
/**
 * "Touching" an object allows you to mark an item in the view as "updated",
 * even if the object itself wasn't directly updated.
 *
 * This is most often useful when a view is being used by a tableView,
 * but the tableView cells are also dependent upon another object in the database.
 *
 * For example:
 *
 *   You have a view which includes the departments in the company, sorted by name.
 *   But as part of the cell that's displayed for the department,
 *   you also display the number of employees in the department.
 *   The employee count comes from elsewhere.
 *   That is, the employee count isn't a property of the department object itself.
 *   Perhaps you get the count from another view,
 *   or perhaps the count is simply the number of keys in a particular collection.
 *   Either way, when you add or remove an employee, you want to ensure that the view marks the
 *   affected department as updated so that the corresponding cell will properly redraw itself.
 *
 * So the idea is to mark certain items as "updated" (in terms of this view) so that
 * the changeset for the view will properly reflect a change to the corresponding index.
 * But you don't actually need to update the item on disk.
 * This is exactly what "touch" does.
 *
 * Touching an item has very minimal overhead.
 * It doesn't cause the groupingBlock or sortingBlock to be invoked,
 * and it doesn't cause any writes to the database.
 *
 * You can touch
 * - just the object
 * - just the metadata
 * - or both object and metadata (the row)
 * 
 * If you mark just the object as changed,
 * and neither the groupingBlock nor sortingBlock depend upon the object,
 * then the view doesn't reflect any change.
 * 
 * If you mark just the metadata as changed,
 * and neither the groupingBlock nor sortingBlock depend upon the metadata,
 * then the view doesn't relect any change.
 * 
 * In all other cases, the view will properly reflect a corresponding change in the notification that's posted.
**/

- (void)touchRowForKey:(NSString *)key inCollection:(NSString *)collection;
````

If my tableview populated using view mappings from Departments and displays direct information about an Employee (1-1 relationship). I would like my tableview cell to be updated if related employee is updated.

Would it be possible for relationships extension to notify, or even better: touch source or destination relationships when a node is updated ?

Thanks,

**Comment by robbiehanson** [comment]
Just an update on this:

I've added the YapDatabaseHooks extension to the repository. It is, unfortunately, still missing a corresponding Wiki article. I'll be adding that soon. But the code to do this kind of stuff is now there.

**Comment by jeremiegirault** [comment]
Great, I'll take a look at this, thanks

**Comment by robbiehanson** [comment]
Another update on this:

I've updated the YapDatabaseHooks extension to be much easier to use. It should now be the best way to implement such a feature.

Additionally, I've improved "touch" by expanding upon its functionality. YapDatabaseReadWriteTransaction now has a "touch" method that will propagate the touch to all extensions. Further, YapDatabaseBlockInvoke now standardizes how extensions should respond to the various touch events.
